apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/DmlTests.cs

```diff
@@ -0,0 +1,484 @@
+            string key = CreateTestRows();
+            using (var connection = _fixture.GetConnection())
+            {
+                RetryHelpers.RetryOnce(() =>
+                {
+                    string dml = $"INSERT INTO {_fixture.TableName} (Key, OriginalValue, Value) VALUES (@key, @ov1, @v1), (@key, @ov2, @v2)";
+                    using (var command = connection.CreateDmlCommand(dml))
+                    {
+                        command.Parameters.Add("key", SpannerDbType.String, key);
+                        command.Parameters.Add("ov1", SpannerDbType.Int64, 5);
+                        command.Parameters.Add("v1", SpannerDbType.Int64, 10);
+                        command.Parameters.Add("ov2", SpannerDbType.Int64, 6);
+                        command.Parameters.Add("v2", SpannerDbType.Int64, 20);
+                        Assert.Equal(2, command.ExecuteNonQuery());
+                    }
+                });
+            }
+            var actual = FetchValues(key);
+            var expected = new Dictionary<int, int>
+            {
+                { 0, 0 },
+                { 1, 1 },
+                { 2, 2 },
+                { 3, 3 },
+                { 4, 4 },
+                { 5, 10 }, // New row
+                { 6, 20 }, // New row
+            };
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void InsertWithQuery()
+        {
+            string key = CreateTestRows();
+            using (var connection = _fixture.GetConnection())
+            {
+                RetryHelpers.RetryOnce(() =>
+                {
+                    // Slightly odd choice here: "OriginalValue" has to be unique, so we end up with the inserted Value being the real original value.
+                    string dml = $"INSERT INTO {_fixture.TableName} (Key, OriginalValue, Value) SELECT Key, OriginalValue + 10, Value FROM {_fixture.TableName} WHERE CopyMe AND KEY=@Key";
+                    using (var command = connection.CreateDmlCommand(dml))
+                    {
+                        command.Parameters.Add("key", SpannerDbType.String, key);
+                        Assert.Equal(2, command.ExecuteNonQuery());
+                    }
+                });
+            }
+            var actual = FetchValues(key);
+            var expected = new Dictionary<int, int>
+            {
+                { 0, 0 },
+                { 1, 1 },
+                { 2, 2 },
+                { 3, 3 },
+                { 4, 4 },
+                { 13, 3 }, // Copied
+                { 14, 4 } // Copied
+            };
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void InvalidQuery()
+        {
+            string key = CreateTestRows();
+            using (var connection = _fixture.GetConnection())
+            {
+                // Slightly odd choice here: "OriginalValue" has to be unique, so we end up with the inserted Value being the real original value.
```

Comment:
I think this comment is out of place here.

Reply:
Nope, that's deliberate - basically "original value" doesn't mean what one might expect it to.
Happy to reword the comment though.

Reply:
I just don't understand it in this test. It's in the test above and there it makes sense, but in this one?

Reply:
In other tests, OriginalValue is the value the row was created with, and Value is a modified value.

In this case we're *inserting* new rows based on existing rows - so they can't share the same OriginalValue.

Reply:
I think you are referring to the above test, the one in line 189 called InsertWithQuery, and there I get the comment.
But it being here in this test called InvalidQuery, in which you are just using an invalid query to test that it throws, I think makes no sense. I think this was just a copy paste thing?

Reply:
Aha! Yes, sorry. Yup, this was indeed just a C&P fail. I'll remove the comment.